node/toolrunner.ts

```diff
@@ -152,38 +153,36 @@ export class ToolRunner extends events.EventEmitter {
         let toolPath: string = this._getSpawnFileName();
         let args: string[] = this._getSpawnArgs(options);
         let cmd = noPrefix ? '' : '[command]'; // omit prefix when piped to a second tool
+        let commandParts: string[] = [];
         if (process.platform == 'win32') {
             // Windows + cmd file
             if (this._isCmdFile()) {
-                cmd += toolPath;
-                args.forEach((a: string): void => {
-                    cmd += ` ${a}`;
-                });
+                commandParts.push(toolPath);
+                commandParts = commandParts.concat(args);
             }
             // Windows + verbatim
             else if (options.windowsVerbatimArguments) {
-                cmd += `"${toolPath}"`;
-                args.forEach((a: string): void => {
-                    cmd += ` ${a}`;
-                });
+                commandParts.push(`"${toolPath}"`);
+                commandParts = commandParts.concat(args);
+            }
+            else if (options.shell) {
```

Comment:
Shell should be added to the IExecOptions interface, right?

Reply:
@damccorm Yes. It was added in #637.

Reply:
Oh right, sorry!